draft-ietf-jsonpath-base.md

```diff
@@ -413,51 +433,70 @@ the grammar in this document, its UTF-8 form SHOULD first be decoded into
 Unicode code points as described
 in {{RFC3629}}.
 
-
-## Terminology
-
-A JSON value is logically a tree of nodes.
-
-Each node holds a JSON value (as defined by {{RFC8259}}) of one of the
-types object, array, number, string, or one of the literals `true`,
-`false`, or `null`.
-The type of the JSON value held by a node is
-sometimes referred to as the type of the node.
-
-
-## Implementation
-
-An implementation of this specification, from now on referred to simply as
-"an implementation", SHOULD takes two inputs, a JSONPath and a JSON value,
-and produce
-a possibly empty list of nodes of the JSON value which are selected by
-the JSONPath or an error (but not both).
-
-If no node is selected and no error has occurred, an implementation MUST
-return an empty list of nodes.
-
-Syntax errors in the JSONPath SHOULD be detected before selection is attempted
-since these errors do not depend on the JSON value.
-Therefore, an implementation SHOULD take a JSONPath and produce an optional
-syntax error and then,
-if and only if an error was not produced, SHOULD take a JSON value and
-produce a list of nodes or an error (but not both).
-
-Alternatively, an implementation MAY take a JSONPath and a JSON value
-and produce a list of nodes or an optional error (but not both).
-
-For any implementation, if a syntactically invalid JSONPath is provided,
-the implementation MUST return an error.
-
-If a syntactially invalid JSON value is provided, any implementation SHOULD
-return an error.
+A string to be used as a JSONPath query needs to be *well-formed* and
+*valid*.
+A string is a well-formed JSONPath query if it conforms to the syntax
+of JSONPath.
+A well-formed JSONPath query is valid if it also fulfills all semantic
+requirements posed by this document.
+
+Well-formedness and validity of JSONPath queries are independent of
+the JSON value the query is applied to; no further errors can be
+raised during application of the query to a JSON value.
+
+(Obviously, an implementation can still fail when executing a JSONPath
+query, e.g., because of resource depletion, but this is not modeled in
+the present specification.)
+
+> Discussion: are we ready for the error model defined here, i.e. one
+> that does not have runtime errors, but only compile time errors?
```

Comment:
I think this needs further discussion before representing it in a PR.

Reply:
Indeed.  So we could resolve this question before merging, or we could merge with the question intact, to be resolved in later PR.

Reply:
Are you saying that we should leave this markdown block quote in place to spur discussion?

Reply:
Yes.  Until that discussion is resolved.

Reply:
If that's the convention we want to follow for authoring the spec, this is fine.